pkg/cloud/externalconn/connectionpb/connection.proto

```diff
@@ -0,0 +1,41 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+syntax = "proto3";
+package cockroach.cloud.externalconn.connectionpb;
+option go_package = "connectionpb";
+
+import "gogoproto/gogo.proto";
+import "roachpb/api.proto";
+
+enum ConnectionType {
+  option (gogoproto.goproto_enum_prefix) = false;
+
+  UNSPECIFIED = 0 [(gogoproto.enumvalue_customname) = "TypeUnspecified"];
+  STORAGE = 1 [(gogoproto.enumvalue_customname) = "TypeStorage"];
+  KMS = 2 [(gogoproto.enumvalue_customname) = "TypeKMS"];
```

Comment:
KMS is not used anywhere so I've deleted it. Storage is returned by nodelocal. As such the type isn't used anywhere today, but it will be used by `SHOW BACKUP EXTERNAL CONNECTIONS` when we do add that in.